src/grappler/ngraph_optimizer.cc

```diff
@@ -186,7 +186,8 @@ Status NgraphOptimizer::Optimize(tensorflow::grappler::Cluster* cluster,
   }
 
   // 4. Encapsulate clusters then, if requested, dump the graphs.
-  TF_RETURN_IF_ERROR(EncapsulateClusters(&graph, idx));
+  FunctionDefLibrary* fdeflib_new = new FunctionDefLibrary();
+  TF_RETURN_IF_ERROR(EncapsulateClusters(&graph, idx, fdeflib_new));
```

Comment:
In this case, where we are not freeing the fdef should there be someplace where should be freeing it. One place i can think of is in the destructor of nGraphEncapsulate Op.

Reply:
Added comment why it does not need a free: According to the docs, the message assumes ownership and is responsible for clean up. a free on our part will probably trigger 2 frees.